File: typescript/src/services/api-keys-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CreateApiKeyDto, PaginationOptions, RevokeApiKeyDto, UpdateApiKeyDto } from '../models/index.js';


export class APIKeysApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getApiKeys(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`api-keys`, request);
  }

  async createApiKey(request: CreateApiKeyDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`api-keys/create`, request);
  }

  async updateApiKey(request: UpdateApiKeyDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`api-keys/update`, request);
  }

  async revokeApiKey(request: RevokeApiKeyDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`api-keys/revoke`, request);
  }

}

File: c#/src/Docmost.Sdk/Services/APIKeysApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class APIKeysApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetApiKeysAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("api-keys", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreateApiKeyAsync(CreateApiKeyDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("api-keys/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateApiKeyAsync(UpdateApiKeyDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("api-keys/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> RevokeApiKeyAsync(RevokeApiKeyDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("api-keys/revoke", request, cancellationToken).ConfigureAwait(false);
    }

}
